Include/internal/pycore_object.h

```diff
@@ -272,17 +272,93 @@ _Py_DECREF_NO_DEALLOC(PyObject *op)
 }
 
 #else
-// TODO: implement Py_DECREF specializations for Py_GIL_DISABLED build
+
 static inline void
 _Py_DECREF_SPECIALIZED(PyObject *op, const destructor destruct)
```

Comment:
This function is barely used now, but there's a similar function `PyStackRef_CLOSE_SPECIALIZED`.

There are two possible reasons to implement this as something that's not just `Py_DECREF`:

1) Performance, but it's not really clear that it's going to be any faster
2) To avoid `_Py_Dealloc` so that you can enable this assertion https://github.com/python/cpython/blob/6227662ff3bf838d31e9441eda935d24733d705a/Objects/object.c#L3166-L3170, but that means you can't call `_Py_MergeZeroLocalRefcount` because that internally may call `_Py_Dealloc`

I think (2) would be difficult to achieve because of `_Py_brc_queue_object`.